common/app/com/commercetools/sunrise/sessions/wishlist/DefaultWishlistInSession.java

```diff
@@ -33,17 +41,24 @@ public DefaultWishlistInSession(final Configuration configuration, final ObjectS
     }
 
     @Override
+    public Optional<WishlistViewModel> findWishlist() {
+        return session.findObjectByKey(wishlistSessionKey, WishlistViewModel.class);
+    }
+
+    @Override
     protected Logger getLogger() {
         return LOGGER;
     }
 
     @Override
-    protected void storeAssociatedData(@NotNull final ShoppingList value) {
-        session.overwriteValueByKey(wishlistIdSessionKey, value.getId());
+    protected void storeAssociatedData(@NotNull final ShoppingList wishlist) {
+        session.overwriteValueByKey(wishlistIdSessionKey, wishlist.getId());
+        session.overwriteObjectByKey(wishlistSessionKey, wishlistViewModelFactory.create(wishlist));
```

Comment:
Yes, it has nothing to do with wishlist, but before we had a limit of 5-6 items until the session stopped working (and mini-cart was not updated). but if we add wishlist plus the additional information, it will not work with more than 2 items on average. Since I enabled cache in categoryTree, I've considered moving fully to cache. It's actually quite easy, just changing the default implementation of the ObjectStoringStrategy class. But we have to keep it in mind :)